### PR TITLE
TGB-41 Implement comma separated delete map/pool/queue

### PR DESCRIPTION
**Pull request by BhavyaLight**
Update delete_maps to take multiple inputs

### CODE REVIEW

File: src/commands/pools/delete_map.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import {ApplicationCommandOptionTypes} from "discord.js/typings/enums";
-import {ICommand} from "wokcommands";
+import { ApplicationCommandOptionTypes } from "discord.js/typings/enums";
+import { ICommand } from "wokcommands";
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts]: import formatting no spaces around curly braces

File: src/commands/pools/delete_map.ts [hunk 2]
```diff
@@ -16,36 +16,48 @@ export default {
...
+            name: "map_uuids",
+            description: "Comma separated list of uuids of the map to remove",
+            type: ApplicationCommandOptionTypes.STRING,
             required: true,
         },
     ],
 
-    callback: async ({interaction}) => {
-        const {options, channelId, guildId, member} = interaction;
+    callback: async ({ interaction }) => {
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts]: remove spaces around curly braces

File: src/commands/pools/delete_map.ts [hunk 3]
```diff
@@ -16,36 +16,48 @@ export default {
...
             return "This command can only be run in a text channel in a server";
         }
 
-        let guild = await Guild.findOneBy({id: guildId});
+        let guild = await Guild.findOneBy({ id: guildId });
         if (!guild) {
             guild = new Guild(guildId);
         }
-        if(!isAdmin(member as GuildMember, guild)) {
+        if (!isAdmin(member as GuildMember, guild)) {
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts]: no space after `if ()`

File: src/commands/pools/delete_map.ts [hunk 4]
```diff
@@ -16,36 +16,48 @@ export default {
...
+        await interaction.deferReply();
 
         // get the command parameters
-        const mapUuid = ensure(options.getInteger("map_uuid"));
+        const mapUuids = ensure(options.getString("map_uuids")).split(",");
+        const deleteGameResponses: string[] = [];
+        const deleteGamePromises = [];
+
+        for (const mapUuid of mapUuids) {
+            deleteGamePromises.push(deleteGameMap(parseInt(mapUuid), guildId)
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts]: input `1,2s,3` will cause exception here, catch and push response 'map id 2s is invalid'

File: src/commands/pools/delete_map.ts [hunk 5]
```diff
@@ -16,36 +16,48 @@ export default {
...
+                .then(deleteGameResponse => {
+                    deleteGameResponses.push(deleteGameResponse);
+                    interaction.editReply(deleteGameResponses.join("\n"));
+                }));
+        }
 
-        return await deleteGameMap(mapUuid, guildId);
-    },
+        await Promise.all(deleteGamePromises)
+        .then(() => interaction.editReply(deleteGameResponses.join("\n")));
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts:68]: add indent for `.then` and `.catch` (any chained `.xyz` function calls on newlines should be indented)

**Review by Divy1211 — CHANGES_REQUESTED** [review]
`delete_q` and `delete_pool` need to be converted the same way.

File: src/commands/pools/delete_map.ts (hunk 2 above)
Comment (BhavyaLight) [reply src/commands/pools/delete_map.ts]: Done

File: src/commands/pools/delete_map.ts (hunk 3 above)
Comment (BhavyaLight) [reply src/commands/pools/delete_map.ts]: Done

File: src/commands/pools/delete_map.ts (hunk 4 above)
Comment (BhavyaLight) [reply src/commands/pools/delete_map.ts]: Done

File: src/commands/pools/delete_map.ts (hunk 5 above)
Comment (BhavyaLight) [reply src/commands/pools/delete_map.ts:68]: will let the new eslint configs handle the indentation.

File: src/commands/pools/delete_map.ts [hunk 6]
```diff
@@ -42,10 +42,29 @@ export default {
...
+        const mapUuids = ensure(options.getString("map_uuids")).split(",");
+        const deleteGameResponses: string[] = [];
+        const deleteGamePromises = [];
 
-        return await deleteGameMap(mapUuid, guildId);
-    },
+        for (const mapUuid of mapUuids) {
+            // Invalid input was provided by the user
+            if(!Number(mapUuid)){
+                deleteGameResponses.push(`Error : Map with id ${mapUuid} should be a number!`);
```
Comment (Divy1211) [inline src/commands/pools/delete_map.ts]: Response Consistency: No space before colon in `Error: Map with id`...

File: src/commands/pools/delete_pool.ts [hunk 7]
```diff
@@ -42,10 +42,29 @@ export default {
...
-        const poolUuid = ensure(options.getInteger("pool_uuid"));
+        const poolUuids = ensure(options.getString("pool_uuids")).split(",");
+        const deletePoolResponses: string[] = [];
+        const deletePoolPromises = [];
 
-        return await deletePool(poolUuid, guildId);
+        for (const poolUuid of poolUuids) {
+            // Invalid input was provided by the user
+            if(!Number(poolUuid)){
+                deletePoolResponses.push(`Error : Pool with id ${poolUuid} should be a number!`);
```
Comment (Divy1211) [inline src/commands/pools/delete_pool.ts]: Response Consistency: No space before colon in `Error: Pool with id`...

File: src/commands/queues/delete_q.ts [hunk 8]
```diff
@@ -43,10 +43,29 @@ export default {
...
-        const uuid = ensure(options.getInteger("uuid"));
+        const qUuids = ensure(options.getString("uuids")).split(",");
+        const deleteQueueResponses: string[] = [];
+        const deleteQueuePromises = [];
 
-        return await deleteQueue(uuid, channelId);
+        for (const queueUuid of qUuids) {
+            // Invalid input was provided by the user
+            if(!Number(queueUuid)){
+                deleteQueueResponses.push(`Error : Queue with id ${queueUuid} should be a number!`);
```
Comment (Divy1211) [inline src/commands/queues/delete_q.ts]: Response Consistency: No space before colon in `Error: Queue with id`...

**Review by Divy1211 — CHANGES_REQUESTED** [review]
Merge after these 3 minor changes are resolved

**Pull request merged**